BGAnimations/ScreenGameplay overlay/FailOnHoldStart.lua

```diff
@@ -0,0 +1,28 @@
+-- We need a way to check if the player gave up before the song properly ended.
+-- It doesn't look like the engine broadcasts any messages that would be helpful here,
+-- so we do the best we can by checking the song position.
+
+local af = Def.ActorFrame{
+	OffCommand=function(self)
+		local fail = (GAMESTATE:GetCurMusicSeconds() < GAMESTATE:GetCurrentSong():GetLastSecond())
+
+		if GAMESTATE:IsCourseMode() then
+			local course = GAMESTATE:GetCurrentCourse()
+			if GAMESTATE:GetCourseSongIndex() + 1 < course:GetNumCourseEntries() then
+				fail = true
+			end
+		end
+
+		-- It doesn't matter who held start button, we have to fail both players as we
```

Comment:
Change the comment regarding "start button" as that's not longer the condition.

Reply:
Oh I guess technically it is. But how about, "It doesn't matter who ended the song early"

Reply:
done!